content/Rails - Multiple Databases.md

```diff
@@ -0,0 +1,119 @@
+# ** Rails 6 - Multiple Databases **
+
+![main image](images/databases.jpg)
+
+Rails 6.0 came up with this amazing enhancement, but for most people, I'm aware they think that is a new feature.
+In my opinion, both are correct, because the actual state of multiple databases before rails 6.0 was not even useful to consider it a completed feature
+
+Let's dive into Rails < 6 state, Rails 6 introduced changes and the current roadmap
+
+
+## Rails 5 state
+ActiveRecord supports multiple databases, but Rails < 6 doesn't provide a way to manage them.
+As said you needed to handle everything (yes, everything) in a really handy way. Like creating your own tasks for `seeds`, `db:prepare`, `db:migrate`, etc.
+Then you needed to create a sort of generator to handle your secondary database migrations following conventions. And finally (as you haven't done enough custom work), create an initializer to config your connection to the new database.
+
+So yes, ActiveRecord supported multiple databases, but it was a complete and real mess you didn't want to get into unless really necessary.
+
+## Rails 6.0 state
+
+#### New stuff
+
+- Multiple primary databases and a replica for each
+
+This is amazing, now you can easiliy configure it in your `database.yml`  like this
+``` ruby
+development:
+  primary:
+    database: my_primary_database
+    user: root
+    adapter: sqlite3
+  primary_replica:
+    database: my_primary_database
+    user: root_readonly
+    adapter: sqlite3
+    replica: true
+  animals:
+    database: my_animals_database
+    user: animals_root
+    adapter: sqlite3
+    migrations_paths: db/animals_migrate
+  animals_replica:
+    database: my_animals_database
+    user: animals_readonly
+    adapter: sqlite3
+    replica: true
+```
+
+Check how you can define your migration paths for each database, and how seem to have the power to have multiple adapters (still working on a POC for this assumption).
+An example migration with this would be: `rails g migration CreateCats name:string --database animals`
```

Comment:
I would like to see the output of the command, I'm too lazy to write it myself

Reply:
You will get fat as hell at this laziness rate dude 🐷 .... but you will have it